hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -586,19 +587,22 @@ stateQueryClient tracer tipVar utxoVar address =
       check (tip == genesisPoint)
     pure $ LSQ.SendMsgReAcquire Nothing clientStAcquiring
 
-  handleEraMismatch :: m (LSQ.ClientStAcquired Block (Point Block) (Query Block) m ())
-  handleEraMismatch = do
-    -- FIXME: log something before looping back?
+  handleEraMismatch :: MismatchEraInfo (CardanoEras StandardCrypto) -> m (LSQ.ClientStAcquired Block (Point Block) (Query Block) m ())
+  handleEraMismatch (mkEraMismatch -> Ouroboros.EraMismatch{Ouroboros.ledgerEraName, Ouroboros.otherEraName}) = do
+    traceWith tracer $ EraMismatchError ledgerEraName otherEraName
     threadDelay 30
     reset
 
 --
 -- Logs
 --
 
+-- NOTE(AB): We don't use records here because there's no ToJSON insetance for SomePoint so there's no point
+-- in using a Generic ToJSON instance
 data TinyWalletLog
   = InitializingWallet SomePoint (Map TxIn TxOut)
   | ApplyBlock (Map TxIn TxOut) (Map TxIn TxOut)
+  | EraMismatchError Text Text
```

Comment:
Which is which?

Should: Give these things names or re-use `Ouroboros.EraMismatch` type.

Reply:
This comment could equally well apply to the other constructor. I did not want to change too much of this file, only address the`FIXME`.

Reply:
Fair enough. But the introduced change should then not make things worse. I see three options, pick one:
* `type ExpectedEra = Text` and `type ActualEra = Text` and use them
* Add record fields for this constructor, I don't think it's odd to have them not on all of them and I agree, not changing everything is sensible!
* Use the `EraMismatch` type which seems to already distinguish the actual/expected.